portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,42 +35,135 @@
+    constructFromInput(request);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/comment.html");
+  }
 
+  /* Prepare the query */
+  public static PreparedQuery fetch() {
+    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    return datastore.prepare(query);
+  }
 
+  /* Construct the array of Comment objects to manipulate it after */
+  public static List<Comment> toInternal(PreparedQuery results) {
     List<Comment> comments = new ArrayList<>();
+
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
       String text = (String) entity.getProperty("text");
       String author = (String) entity.getProperty("author");
-      Comment comment = new Comment(id, text, author);
+      String mood = (String) entity.getProperty("mood");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String userFilter = (String) entity.getProperty("filter");
+
+      Comment comment = new Comment(id, text, author, mood, timestamp, userFilter);
       comments.add(comment);
     }
 
-    Gson gson = new Gson();
+    return comments;
+  }
 
-    response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(comments));
+  /* Create filters for timestap and text of comment */
+  public static Comparator timestampFilter() {
+    Comparator<Comment> byTimestamp = Comparator.comparingLong(Comment::getTimestamp);
+    return byTimestamp;
   }
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
+  public static Comparator textFilter() {
+    Comparator<Comment> byLength = Comparator.comparingInt(Comment::getTextLength);
+    return byLength;
+  }
+
+  /* Get the filter from the last added comment => that is the filter wanted */
+  public static String getLastFilter(List<Comment> comments) {
+    Collections.sort(comments, timestampFilter().reversed());
+    return comments.get(0).getFilter();
+  }
+
+  /* Filter the comments => sort by timestamp or text */
+  public static void sort(List<Comment> comments, String userFilter) {
+    if ("newest".equals(userFilter)) {
+      Collections.sort(comments, timestampFilter().reversed());
+    } else if ("oldest".equals(userFilter)) {
+      Collections.sort(comments, timestampFilter());
+    } else if ("longest".equals(userFilter)) {
+      Collections.sort(comments, textFilter().reversed());
+    }
+  }
+
+  /* Convert the filtered comments to json */
+  public static String toJson(List<Comment> comments) {
+    Gson gson = new Gson();
+    return gson.toJson(comments);
+  }
+
+  /* Get the input from the form, create entity and add to datastore*/
+  public static void constructFromInput(HttpServletRequest request) {
```

Comment:
This method saves comments into datastore. But `constructFromInput` doesn't suggest any of that. I think you can find a better name for this method.